dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala

```diff
@@ -51,6 +53,36 @@ case class Table(
   def put(hashPK: Any, attributes: (String, Any)*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, attributes: _*)
   def put(hashPK: Any, rangePK: Any, attributes: (String, Any)*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, rangePK, attributes: _*)
 
+  def put(entity: Product)(implicit dynamoDB: DynamoDB) = {
+    val attrs = getAttrValuesToList(entity)
+    if(attrs("keys").exists(x => x._1 == "hashPK") && attrs("keys").exists(x => x._1 == "rangePK")) {
+      val hashPK = attrs("keys").find(f => f._1 == "hashPK").get._2
+      val rangePK = attrs("keys").find(f => f._1 == "rangePK").get._2
+      dynamoDB.put(this, hashPK, rangePK, attrs("attributes"): _*)
+    }
+    else if (attrs("keys").exists(x => x._1 == "hashPK")) {
+      val hashPK = attrs("keys").find(f => f._1 == "hashPK").get._2
+      dynamoDB.put(this, hashPK, attrs("attributes"): _*)
+    }
+  }
+
+  private def getAttrValuesToList(entity: Product) = {
+    val fields = entity.productElementNames.toList
+    val f1 = fields.map(f => f -> {
+      val fv = entity.getClass.getDeclaredField(f)
+      fv.setAccessible(true)
```

Comment:
@jwszolek Specifically, accessing a possibly private field this way is not acceptable even you use Product in Scala. Code using `setAccessible` in **Java reflection API** can be easily broken. 

Can you explore alternatives here? I haven't checked if the same way works for this change but just for your information: https://github.com/skinny-framework/skinny-framework/blob/3.1.0/framework/src/main/scala/skinny/controller/feature/RequestScopeFeature.scala#L208

Reply:
hey @seratch , thanks for the link to skinny-framework. Good inspiration. Just added getterNamesFromEntity() that should fix the issue you mentioned. Let me know if something else in the code needs to be revisited.